src/NuGet.Core/NuGet.Protocol/Resources/MetadataResourceV3.cs

```diff
@@ -131,5 +120,34 @@ public MetadataResourceV3(HttpSource client, RegistrationResourceV3 regResource)
 
             return results;
         }
+
+        public override async Task<IEnumerable<NuGetVersionWithDeprecationInfo>> GetVersionsAndDeprecationInfo(
```

Comment:
How about changing the dependency from MetadataResource to PackageMetadataResource? 

That one has a API like the following: https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Protocol/Resources/PackageMetadataResource.cs#L16 
where you can get the deprecation information.

Reply:
The issue is that `IPackageSearchMetadata` does not contain `Version` information

Reply:
what about PackageIdentity?

Reply:
Overlooked that one! ty!

In `MsBuildApiUtility` lines 443 and 456 we need to construct the `IPackageSearchMetadata` ourselves then as the `ResolvedVersion` property will be of type `IPackageSearchMetadata`.

Would the following piece of code be OK to construct it?

```
ResolvedVersion = PackageSearchMetadataBuilder.FromIdentity(
    new PackageIdentity(library.Name, library.Version)).Build()
```

See https://github.com/NuGet/NuGet.Client/pull/2928/commits/696eb89e68a24efc36e917463c19a11e4cd1335c

Reply:
Yeah, I think that's fine.